_posts/2018-01-01-futures-0.3.0-alpha.3.md

```diff
@@ -0,0 +1,71 @@
+---
+layout: post
+title:  "Futures 0.3.0-alpha.3"
+author: "Josef Brandl"
+author_github: "MajorBreakfast"
+date:   2018-01-01
+categories: blog
+---
+
+This release introduces the easy to use `spawn!` and `spawn_with_handle!` macros for use in async functions, closures and blocks. Both let you spawn a task that polls the given future to completion. The task is spawned onto the current context's executor.
+
+`spawn!` lets you spawn futures with `Output = ()`:
+
+```rust
+#![feature(async_await, await_macro, futures_api)]
+#[macro_use] extern crate futures;
+
+spawn!(future).unwrap();
+```
+
+`spawn_with_handle!` lets you spawn futures with any `Output` type and also lets you await the output:
+
+```rust
+let join_handle = spawn_with_handle!(future).unwrap();
```

Comment:
i don't know if this is necessary in this post, but i would love to have documentation explaining what makes calling `spawn_with_handle!` and awaiting on the handle different than awaiting the future directly, with usecases explaining when to use one or the other.

Reply:
`spawn_with_handle!` uses a oneshot channel to send the value back. `spawn!` doesn't do that. Hence, it's a bit cheaper.

Reply:
I think @tinaun was referring to why would you choose one or the other of the following:

```rust
let value = await!(spawn_with_handle!(future).unwrap());
let value = await!(future);
```

I think the answer is that you would always choose the second, `spawn_with_handle!` only really makes sense to use if you are then going to do something concurrently, e.g.

```rust
let first_thing = spawn_with_handle!(future).unwrap();
let second_thing = await!(other_future);
let first_thing = await!(first_thing);
```

I agree that it would be good to include this in the post somehow.